server/remote_cache/cachetools/cachetools.go

```diff
@@ -269,6 +276,235 @@ func UploadProtoToCAS(ctx context.Context, cache interfaces.Cache, instanceName
 	return uploadProtoToCache(ctx, cas, instanceName, in)
 }
 
+// BatchCASUploader uploads many files to CAS concurrently, batching small
+// uploads together and falling back to bytestream uploads for large files.
+type BatchCASUploader struct {
+	ctx              context.Context
+	eg               *errgroup.Group
+	cache            interfaces.Cache
+	bsClient         bspb.ByteStreamClient
+	casClient        repb.ContentAddressableStorageClient
+	req              *repb.BatchUpdateBlobsRequest
```

Comment:
nit: Can we give this a more descriptive name to convey that this is the request that is being built up for the batch upload? Maybe something like pendingBulkReq or unsentBulkReq?

Reply:
Went with `currentUnsentBatchReq`